backend/app.py

```diff
@@ -146,6 +156,55 @@ def json(self):
             'Role_ID': self.Role_ID,
             'Applied': self.Applied
         }
+################ login endpoints ##################################################
+
+# for staff to login
+@app.route('/login', methods=['POST'])
+def login():
+    try: 
+        data = request.get_json()
+        email = data['Email']
+        password = data['Password']
+        access_rights = data['Access_Rights']
+
+        staff = Staff.query.filter_by(Email=email).first()
+        if not staff:
+            return jsonify({'message': 'Staff member not found'}), 404
+
+        if staff.Password != password:
+            return jsonify({'message': 'Incorrect password'}), 401
+
+        if access_rights == 'HR':
+            access_rights = 1
+        elif access_rights == 'Staff':
+            access_rights = 2
+
+        #staff.Access_Rights = 1 can access HR and Staff pages
+        #staff.Access_Rights = 2 can only access Staff pages
+
+        if access_rights == 1 and staff.Access_Rights != 1:
+            return jsonify({'message': 'Restricted Access'}), 401
+
+        
+        session['staff_id'] = staff.Staff_ID
+        session['access_rights'] = staff.Access_Rights
+
+        response_data = {
+            'staff_id': staff.Staff_ID,
+            'Access_Rights': staff.Access_Rights,
+            'Country': staff.Country,
+            'Dept': staff.Dept,
+            'Email': staff.Email,
+            'Password': staff.Password,
+            'Staff_FName': staff.Staff_FName,
+            'Staff_LName': staff.Staff_LName
+        }
+
+        return jsonify(response_data), 200
+    
+    except Exception as e:
+        return jsonify({'message': str(e)}), 500    
+
```

Comment:
could you add in a logout endpoint such that the flask_session is deleted once a user closes the browser or logs out so that the flask_session will not be stored?

<img width="184" alt="image" src="https://github.com/L1uM4rt1n/SPM-Project/assets/96768015/47f2417b-d1bc-40c6-aa4c-37d9c3e7d122">